src/streaming/s3.connection.hh

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include <miniocpp/client.h>
+
+#include <condition_variable>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <span>
+#include <vector>
+
+namespace zarr {
+class S3Connection
+{
+  public:
+    explicit S3Connection(const std::string& endpoint,
+                          const std::string& access_key_id,
```

Comment:
You end up needing to create `std::string`s in the constructor from these, because `minio::s3::BaseUrl` and `minio::creds::StaticProvider` won't take `std::string_view` and it appears there's no implicit conversion.